### PR TITLE
PDFR-21935: Update Terraform Versions

### PR DESCRIPTION
**Pull request by VSpike**
Update to the latest patch version of each minor series. Also, remove
unused older versions as they will just bloat the image.

### CODE REVIEW

**Review by DragosDumitrache — APPROVED** [review]
Guessing this will require quite changes to the atlantis.yaml file in anu as well, right?

**Comment by VSpike** [comment]
> Guessing this will require quite changes to the atlantis.yaml file in anu as well, right?

As far as I can tell, Atlantis will download any version of Terraform you reference. This just decides which versions are rolled into the container.

**Comment by rnaveiras** [comment]
> As far as I can tell, Atlantis will download any version of Terraform you reference. This just decides which versions are rolled into the container.

What do you mean by download here? We bake these versions in the container, and then Atlantis consumes these versions base on the configuration.

I think what @DragosDumitrache is saying is that we need to deploy the new sha at the same time that you get in the changes in the configuration yaml.

As you removed the old versions instead of keeping that change for later, you make this change more complicated than it needs to be, by require synchronization now between the new container being deployed + atlantis.yaml with the relevant references to the versions. 

In a nutshell, deploying this or https://github.com/gocardless/anu/pull/19979 is going to case errors and some planes are going to fail...

**Comment by VSpike** [comment]
@rnaveiras Atlantis will download any version of Terraform on demand that is required, so it's not a problem. Bundling versions into the container just saves downloads for your preferred versions.

```
2023/01/23 15:36:32+0000 [INFO] gocardless/anu#20008: Could not find terraform version 0.12.29 in PATH or /root/.atlantis/bin, downloading from https://releases.hashicorp.com/
2023/01/23 15:36:33+0000 [INFO] gocardless/anu#20008: Downloaded terraform 0.12.29 to /root/.atlantis/bin/terraform0.12.29
2023/01/23 15:36:35+0000 [INFO] gocardless/anu#20008: Successfully ran "/root/.atlantis/bin/terraform0.12.29 init -input=false -no-color -upgrade" in "/root/.atlantis/repos/gocardless/anu/20008/prd-staging/terraform/google/projects/business_intelligence_personal_data"
2023/01/23 15:36:35+0000 [INFO] gocardless/anu#20008: Successfully ran "/root/.atlantis/bin/terraform0.12.29 workspace show" in "/root/.atlantis/repos/gocardless/anu/20008/prd-staging/terraform/google/projects/business_intelligence_personal_data"
2023/01/23 15:37:02+0000 [INFO] gocardless/anu#20008: Successfully ran "/root/.atlantis/bin/terraform0.12.29 plan -input=false -refresh -no-color -out \"/root/.atlantis/repos/gocardless/anu/20008/prd-staging/terraform/google/projects/business_intelligence_personal_data/prd-staging.tfplan\" -var-file prd-staging.tfvars -parallelism 100" in "/root/.atlantis/repos/gocardless/anu/20008/prd-staging/terraform/google/projects/business_intelligence_personal_data"
2023/01/23 15:37:21+0000 [INFO] server: Parsed comment as command="apply" verbose=false dir="" workspace="" project="" flags=""
2023/01/23 15:37:23+0000 [INFO] gocardless/anu#20008: Pull request mergeable status: true
2023/01/23 15:37:24+0000 [INFO] gocardless/anu#20008: Starting apply
2023/01/23 15:37:25+0000 [INFO] gocardless/anu#20008: Successfully ran "/root/.atlantis/bin/terraform0.12.29 apply -input=false -no-color \"/root/.atlantis/repos/gocardless/anu/20008/lab/terraform/google/projects/business_intelligence_personal_data/lab.tfplan\"" in "/root/.atlantis/repos/gocardless/anu/20008/lab/terraform/google/projects/business_intelligence_personal_data"
2023/01/23 15:37:25+0000 [INFO] gocardless/anu#20008: Apply successful, deleting planfile
